atrium-api/src/types/string.rs

```diff
@@ -436,6 +452,27 @@ impl Tid {
         }
     }
 
+    /// Construct a new timestamp with the specified clock ID.
+    ///
+    /// Clock IDs 0-31 can be used as an ad-hoc clock ID if you are not concerned
+    /// with this parameter.
+    pub fn from_datetime(cid: u32, time: chrono::DateTime<chrono::Utc>) -> Self {
```

Comment:
And what would be the appropriate type for the clock identifier?

This value is only used for the lower 10 bits, and `atrium_api::types::integer` defines a type that takes into account various bounds. I think `LimitedU32<1023>` would be appropriate here, what do you think?